File: Queues - Python/QueueAsArray.py
class QueueAsArray:

    # Init the variables for the queue
    def __init__(self, capacity=6):
        self.n = capacity
        self.queue = [0] * self.n
        self.front = -1
        self.rear = -1
        self.current_items = 0

    # Inserts a randomly given number into the queue
    def enqueue(self, item):
        if self.is_full():
            print("The Queue is full")
        elif self.is_empty():
            self.front = 0
            self.rear = 0
            self.queue[self.rear] = item
            self.current_items += 1
        else:
            self.rear += 1
            self.queue[self.rear] = item
            self.current_items += 1

    # Deletes the randomly given number from the queue
    def dequeue(self):
        if self.is_empty():
            print("The Queue is empty")
        elif self.front == self.rear:
            self.front = -1
            self.rear = -1
        else:
            self.front += 1
            self.current_items -= 1

    # Displays the queue
    def display(self):
        if self.is_empty():
            print("The Queue is empty")
        else:
            result = "["
            current = self.front
            while True:
                result += str(self.queue[current])
                if current == self.rear:
                    break
                result += ", "
                current = (current + 1) % self.n
            result += "]"
            print(result)

    # Shows the first Item in the queue
    def peek_front(self):
        if self.is_empty():
            print("The Queue is empty")
        else:
            print(" ")
            print(self.queue[self.front])

    # Tells if the queue is full
    def is_full(self):
        return self.current_items == self.n

    # Tells if the queue is empty
    def is_empty(self):
        return self.current_items == 0

    # Shows the current items in the queue
    def get_size(self):
        return self.current_items


# the main method
if __name__ == "__main__":
    queue = QueueAsArray(6)

    queue.enqueue(5)
    queue.enqueue(17)
    queue.enqueue(3)
    queue.enqueue(3)

    queue.dequeue()

    queue.enqueue(13)
    queue.enqueue(11)

    queue.peek_front()
    print(queue.get_size())
    queue.display()

# Note that in Python arrays are handled similar to lists

File: Queues-Explained-main/Queues - Java/QueueAsArray.java
public class QueueAsArray {

    /**
     * Init a default capacity for the queue
     */
    private static final int n = 6;

    /**
     * Init the queue and giving it the declared default capacity
     */
    private int[] queue = new int[n];

    /**
     * Front of the given queue
     */
    private int front = -1;

    /**
     * Rear of the given queue
     */
    private int rear = -1;

    /**
     * The current items in the queue
     */
    private int currentItems = 0;

    /**
     * Inserts a randomly given number into the queue
     * @param item element to be added into the queue
     */
    public void enqueue(int item) {
      if(isFull()) System.out.println("The Queue is full");
      // If the queue is empty, direct front and rear to 0 and insert the item
      // Increase currentItems
      else if(isEmpty()) {
          front = 0;
          rear = 0;
          queue[rear] = item;
          currentItems++;
      } else {
          // If neither is true, increase rear and insert the item
          // Increase currentItems
          rear++;
          queue[rear] = item;
          currentItems++;
      }
    }

    /**
     * Deletes the randomly given number from the queue
     */
    public void dequeue() {
        if(isEmpty()) System.out.println("The Queue is empty");
        // if front is equal to rear, direct both of them to -1
        else if(front == rear) {
            front = -1;
            rear = -1;
          // If neither is true, increase the front (deletes the item)
          // Decrease currentItems
        } else {
            front++;
            currentItems--;
        }
    }

    /**
     * Displays the queue
     */
    public void display() {
        if (isEmpty()) System.out.println("The Queue is empty");
        else {
            StringBuilder result = new StringBuilder("[");
            int current = front;

            while (true) {
                result.append(queue[current]);

                if (current == rear) {
                    break;
                }

                result.append(", ");
                current = (current + 1) % n;
            }

            result.append("]");
            System.out.println(result);
        }
    }

    /**
     * Shows the first Item in the queue
     */
    public void peekFront() {
        if(isEmpty()) System.out.println("The Queue is empty");
        else {
            System.out.println(" ");
            System.out.println(queue[front]);
        }
    }

    /**
     * Tells if the queue is full
     *
     * @return if the current items in the queue are equal to the size n
     */
    public boolean isFull() {
        return currentItems == n;
    }

    /**
     * Tells if the queue is empty
     *
     * @return if there are no current items in the queue
     */
    public boolean isEmpty() {
        return currentItems == 0;
    }

    /**
     * Shows the current items in the queue
     *
     * @return gives back the items
     */
    public int getSize() {
        return currentItems;
    }
}
